ivadomed/loader/patch_filter.py

```diff
@@ -0,0 +1,55 @@
+import numpy as np
+
+
+class PatchFilter(object):
+    """Filter 2D patches from dataset.
+
+    If a patch does not meet certain conditions, it is discarded from the dataset at training time.
+
+    Args:
+        filter_empty_mask (bool): If True, 2D patches where all voxel labels are zeros are discarded at training time.
+        filter_absent_class (bool): If True, 2D patches where all voxel labels are zero for one or more classes are
+            discarded at training time.
+        filter_empty_input (bool): If True, 2D patches where all voxel intensities are zeros are discarded
+            at training time.
+        is_train (bool): Indicates if at training time.
+
+    Attributes:
+        filter_empty_mask (bool): If True, 2D patches where all voxel labels are zeros are discarded at training time.
+            Default: False.
+        filter_absent_class (bool): If True, 2D patches where all voxel labels are zero for one or more classes are
+            discarded at training time. Default: False.
+        filter_empty_input (bool): If True, 2D patches where all voxel intensities are zeros are discarded
+            at training time. Default: False.
+        is_train (bool): Indicates if at training time.
+
+    """
+
+    def __init__(self, filter_empty_mask=False,
+                 filter_absent_class=False,
+                 filter_empty_input=False,
+                 is_train=False):
+        self.filter_empty_mask = filter_empty_mask
+        self.filter_absent_class = filter_absent_class
+        self.filter_empty_input = filter_empty_input
+        self.is_train = is_train
+
+    def __call__(self, sample):
```

Comment:
can we add a type hint for this `sample` here? Is it a dataframe or something? I wasn't super sure.

Reply:
Good point. It's a dict from [here](https://github.com/ivadomed/ivadomed/blob/29f24072c6d6b374f7c23292881cf1a44c10ae00/ivadomed/loader/mri2d_segmentation_dataset.py#L164).
I'll update it here and in slice filter as well.

Reply:
I added docstrings and type hint in d19127b for patch filter and slice filter.